Source/Core/InputCommon/ControllerEmu/ControlGroup/IMUGyroscope.h

```diff
@@ -23,7 +23,7 @@ class IMUGyroscope : public ControlGroup
 
   StateData GetRawState() const;
   // Also updates the state by default
-  std::optional<StateData> GetState(bool update = true);
+  std::optional<StateData> GetState();
```

Comment:
If you come across some code that does something and you don't understand why / it seems meaningless, your should first do a git blame of said code's history to search for an explanation.

https://github.com/dolphin-emu/dolphin/commit/379ffc268d30c66c9cc43a15998369de769efaf0

Pinging @Filoppi to ask if this code scaffolding is still necessary for ongoing or future work.

Reply:
I don't know, theoretically I still have a PR with many fixes and improvements that would make use of this, and ideally I'll finish it one day, but it's hard to guarantee.
Tbh I don't see the point in removing this flag, it's irrelevant for performance and doesn't really hurt. It feels like this is changing the code for the sake of it (at least for this specific change).

Reply:
> Tbh I don't see the point in removing this flag, it's irrelevant for performance and doesn't really hurt. 

I like code which justifies its existence, either for performance, feature or readability reasons. Not being harmful enough to remove is not a good reason to me.

> It feels like this is changing the code for the sake of it

The reason for changing this is for readability. See [Martin Fowler's article about flag arguments](https://martinfowler.com/bliki/FlagArgument.html) for more details

Reply:
It'd be good to make it an enum for sure

Reply:
I only use enums when there are more than two code paths.

What values should the enum contain?

Reply:
None, ForceUpdate? Something like that? Thanks.